doc/development/adding_templates.rst

```diff
@@ -0,0 +1,91 @@
+How to add a new template
+-------------------------
+
+The following steps will help you to add your favourite circuit ansatz to
+PennyLane's :mod:`template <pennylane.template>` library.
+
+PennyLane internally distinguishes :ref:`different types <intro_ref_temp>` of templates, such as
+:mod:`embeddings <pennylane.templates.embeddings>`, :mod:`layers <pennylane.templates.layers>`,
+:mod:`state_preparations <pennylane.templates.state_preparations>` or
+:mod:`soubroutines <pennylane.templates.soubroutines>`.
+Here we will use the template ``MyNewTemplate`` as an example, and you need to replace ``<templ_type>`` with the
+correct template type.
+
+1. **Add the template** by adding a new file ``my_new_template.py`` to the correct ``templates/<templ_type>/``
+   subdirectory, which contains your new template of the form
+
+   .. code-block:: python
+
+        @template
+        def MyNewTemplate(*args, **kwargs):
+            ...
+
+   A template is a function that defines a sequence of quantum gates (without measurements).
+   Since the template is called within a :ref:`quantum function <intro_vcirc_qfunc>`,
+   it can only contain information processing that is allowed
+   inside a quantum functions.
+
+   Make sure you consider the following:
+
+   * To be consistent with other quantum operations (which are classes), the function name (i.e., ``MyNewTemplate``) is
+     capitalized. Choose the name carefully. Good names tell the user what a template is good for, or what architecture
+     it implements.
+
+   * A template is "registered" by using the :mod:`template <pennylane.template>` decorator ``@template``.
+     This allows us to record the queue of operations of a template, which is very useful for testing:
+
+     .. code-block:: python
+
+        with qml.utils.OperationRecorder() as rec:
+            MyNewTemplate(...)
+
+        list_of_gates = rec.queue
+
+   * Your template should have a positional argument called ``wires``, which takes an integer (single wire) or a list
+     of wire indices that the template acts on.
```

Comment:
An example here would be good

Reply:
Actually, I am so unsure if this is even true. In principle, one does not need this argument, but in the past it has been a convention. Deleted it for now.